mapcomposer/app/static/externals/gxp/src/script/plugins/ChartReporting.js

```diff
@@ -48,7 +57,42 @@ gxp.plugins.ChartReporting = Ext.extend(gxp.plugins.Tool, {
             msg: "Not yet implemented"}
          );
     },
-    
+
+    init : function(target) {
+        this.target = target;
+        gxp.plugins.ChartReporting.superclass.init.call(this, target);   
+        if(!this.geoStoreClient) {
+            this.geoStoreClient = new gxp.plugins.GeoStoreClient({
+                url: (this.geostoreUrl) ? this.geostoreUrl : this.target.geoStoreBaseURL,
+                user: (this.geostoreUser) ? this.geostoreUser : this.target.geostoreUser,
```

Comment:
User and Password have to be retrieved from the localStorage. 
Proxy should to be retrieved from the target. Anyway make you sure that the GeoStoreClient uses it **only** if the domain name is different. if not,  your current solution could be better.